app/API/Upload.php

```diff
@@ -0,0 +1,873 @@
+<?php
+
+namespace Gazelle\API;
+use Gazelle\Torrent\EditionInfo;
+use Gazelle\Torrent\TorrentSlot;
+use Gazelle\Util\Time;
+
+ini_set('max_file_uploads', 100);
+define('MAX_FILENAME_LENGTH', 255);
+
+class Upload extends AbstractAPI {
+
+    public function run() {
+        return $this->uploadTorrent();
+    }
+
+    private function uploadTorrent() {
```

Comment:
Initially tried to do that but in the current state of `upload_handler` its not possible. The only way to have a common codebase for these would be to refactor the `upload_handler`. 

Also i'm not sure about this, but it could be because of two different codebases being used. (Whatcd and opusnet).
The way some of the global parameters handled in api and the original side codebase seems to be different.
`$this->db` => This is how DB object is accessible in api, while in the original `upload_handler` code its different. (`$DB->query("`)

Reply:
This method is everything from upload_handler copied, and refactored to work with the API setup.

Reply:
They are same actually.
It may indeed be more complicated, but it is necessary to refactor `upload_handler.php`, otherwise it will not be easy to maintain in the future.

Reply:
Can you help with refactor `upload_handler.php`, if not, I can deal with it.

Reply:
My initial attempt was to accept the data from the api, and just reuse the `upload_handler` class itself.
But that file is not written in a modular way. hence i decided to copy the contents and refactor the code to work with the api while still maintaining the core logic.

Honestly refactoring that file would be a tedious and difficult task for me since i'm not familiar with GPW codebase and its architecture. 

You must have realised by now, that i've been writing my own methods instead of reusing existing one, not know what the G what, didn't realise that the framework actually had transactions etc.

You seem like the best person for that job :)

Reply:
What i've learned from this api framework is that
- `public/api.php` is the entry point and from there control goes to
- `sections/api/index.php` which decides what the user intends to do and picks one of the action file from
- `app/api`, where all the api classes are children of the AbstractAPI